src/main/scala/org/clulab/reach/extern/export/fries/FriesOutput.scala

```diff
@@ -198,115 +196,119 @@ class FriesOutput extends JsonOutputter {
-                               allMentions:Seq[Mention],
-                               passageMap:Map[String, FriesEntry],
-                               contextIdMap:CtxIDed,
-                               startTime:Date,
-                               endTime:Date,
-                               otherMetaData:Map[String, String]): (PropMap, IDed) = {
+
+  /** Return an Assembly Model, representing selected assembly information for this paper. */
+  private def assemblyToModel (paperId: String,
+                               mentions: Seq[Mention],
+                               passageMap: Map[String, FriesEntry],
+                               entityMap: IDed,
+                               eventMap: IDed,
+                               startTime: Date,
+                               endTime: Date,
+                               otherMetaData: Map[String, String],
+                               assemblyApi: Assembler): PropMap = {
     val model:PropMap = new PropMap
     addMetaInfo(model, paperId, startTime, endTime, otherMetaData)
-    val entityMap = new IDed
 
-    // this stores all frames in this output
     val frames = new FrameList
     model("frames") = frames
 
-    // dereference all coreference mentions:
-    val derefedMentions = allMentions.map(m => m.antecedentOrElse(m))
+    for (mention <- eventMap.keys.toList.sortBy(m => (m.sentence, m.tokenInterval.start))) {
+      val fromId = lookupMentionId(mention, entityMap, eventMap)
+      if (fromId.isDefined) {
+        frames ++= make1toMLinkFrames(paperId, mention, fromId.get,
+                                      assemblyApi.getEquivalenceLinks(mention),
+                                      "equivalent-to", passageMap, entityMap, eventMap)
+        frames ++= make1to1LinkFrames(paperId, mention, fromId.get,
+                                      assemblyApi.getCausalPredecessors(mention),
+                                      "preceded-by", passageMap, entityMap, eventMap)
+      }
+    }
+    model                              // return the constructed assembly model
+  }
 
-    for(mention <- derefedMentions) {
+
+  /** Return an Entity Model, representing all entity mentions extracted from this paper. */
+  private def entitiesToModel (paperId: String,
+                               mentions: Seq[Mention],
+                               passageMap: Map[String, FriesEntry],
+                               entityMap: IDed,
+                               contextIdMap: CtxIDed,
+                               startTime: Date,
+                               endTime: Date,
+                               otherMetaData: Map[String, String]): PropMap = {
+    val model:PropMap = new PropMap
+    addMetaInfo(model, paperId, startTime, endTime, otherMetaData)
+
+    val frames = new FrameList
+    model("frames") = frames
+
+    for (mention <- mentions) {
       mention match {
         case em:BioTextBoundMention =>
```

Comment:
Forgive my ignorance, but is this method meant to handle any kind of Entity?  Does the outputter consider a complex to be an entity?  If so, maybe you want to match on the Mention's label, rather than the type of the Mention.